rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/fragment/CovidOtherClinicalDataFragment.java

```diff
@@ -0,0 +1,71 @@
+package io.ona.rdt.fragment;
+
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import androidx.fragment.app.Fragment;
+import io.ona.rdt.R;
+import io.ona.rdt.contract.CovidOtherClinicalDataFragmentContract;
+import io.ona.rdt.domain.Patient;
+import io.ona.rdt.presenter.CovidOtherClinicalDataFragmentPresenter;
+import io.ona.rdt.util.Constants;
+import io.ona.rdt.util.CovidConstants;
+
+public class CovidOtherClinicalDataFragment extends Fragment implements CovidOtherClinicalDataFragmentContract.View,
+        View.OnClickListener {
+
+    private CovidOtherClinicalDataFragmentPresenter presenter;
+    private Patient currPatient;
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        getActivity().findViewById(R.id.patient_profile_tabbed_fragment_container).setVisibility(View.GONE);
+        getActivity().findViewById(R.id.patient_profile_fragment_container).setVisibility(View.VISIBLE);
+
+        presenter = new CovidOtherClinicalDataFragmentPresenter(this);
+        currPatient = getArguments().getParcelable(Constants.FormFields.PATIENT);
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        View rootLayout = inflater.inflate(R.layout.fragment_covid_other_clinical_data, container, false);
+        addListeners(rootLayout);
+        return rootLayout;
+    }
+
+    @Override
+    public void onDestroyView() {
+        getActivity().findViewById(R.id.patient_profile_fragment_container).setVisibility(View.GONE);
+        getActivity().findViewById(R.id.patient_profile_tabbed_fragment_container).setVisibility(View.VISIBLE);
+        super.onDestroyView();
+    }
+
+    @Override
+    public void onClick(View v) {
+        presenter.launchForm(getActivity(), getFormName(v), currPatient);
+    }
+
+    private void addListeners(final View rootLayout) {
+        rootLayout.findViewById(R.id.tv_covid_xray).setOnClickListener(this);
+        rootLayout.findViewById(R.id.tv_covid_wbc).setOnClickListener(this);
+    }
+
+    private String getFormName(View view) {
+        String formName = null;
+        switch (view.getId()) {
```

Comment:
is Java switch efficient? This could be better as a map, also would be more concise

I guess somewhere outside here as a constant?

```java
Map<String, String> viewIdToForm  = new HashMap<String, String>() {{
    put(R.id.tv_covid_wbc, CovidConstants.Form.WBC_FORM);
    put(R.id.tv_covid_xray, CovidConstants.Form.XRAY_FORM);
}};
```

then here

```java
return viewIdToForm.getOrDefault(view, null)
```

this would be O(1), unless cases are implicitly converted to maps, switch has to be O(N), N is number of case statements.

Edit: yea let's use the map, switches do look through all the values in worst case, https://www.artima.com/underthehood/flow.html

Reply:
Cool, refactored.